Tensile/KernelWriter.py

```diff
@@ -1090,6 +1169,139 @@ def setupNewTile(self, kernel, tensorParametersA, tensorParametersB, isPap, isOp
 
     return kl
 
+  ##############################################################################
+  # noGlobalLoadLoop
+  # Create the No Global Load Loop (NLL)
+  ##############################################################################
+  def noGlobalLoadLoop( self, kernel, tensorParametersA, tensorParametersB, pack ):
+    kl = []
+    pflr     = self.numItersPLR
+    localWriteEndIter = kernel["LoopIters"] - self.numItersPLR - 1
+
+    kl.append(self.comment3(" NoGlobalLoadLoop - Begin"))
+
+    if not self.numItersPLR:
+      if self.enable["Wait"]:
+        kl.append(self.wait(kernel, tensorParametersA, tensorParametersB, -1, 0, -1, "4wait for local write"))
+      if self.enable["Sync"]:
+        kl.append(self.syncThreads(kernel))
+
+    self.perIterLocalWriteCode = self.perIterLocalWriteCodeNGLL
+    self.perIterLocalWriteCanSkip = [ 0 for i in range (kernel["LoopIters"]) ]
+
+    for u in range(0, kernel["LoopIters"]):
+      kl.append(self.comment("iter %u"%u))
+      plrIdx = (u+pflr) % (self.numVgprBuffer+1) % kernel["LoopIters"]
+      localReads = Code.Module()
+      isResetLroIter = (u == localWriteEndIter)
+      isSwapAndResetLwoIter = isResetLroIter
+      isSwapLroIter = isResetLroIter
+      if kernel["ScheduleIterAlg"] == 3:
+        # isSwapLroIter = (u == self.nextLoopLRMfmaIndex//(kernel["MIWaveTile"][0] * kernel["MIWaveTile"][1] * kernel["InnerUnroll"]))
+        isSwapAndResetLwoIter = (u == self.lwEndMfmaIndex//(kernel["MIWaveTile"][0] * kernel["MIWaveTile"][1] * kernel["InnerUnroll"]))
+      extraComment = ""
+      if isResetLroIter:
+        extraComment += " (reset local read pointers iteration) "
+      if isSwapAndResetLwoIter:
+        extraComment += " (swap and reset local write pointers iteration) "
+      if isSwapLroIter:
+        extraComment += " (swap local read pointers iteration) "
+
+      pointerLWCode = Code.Module()
+      pointerLRCode = Code.Module()
+      waitCode = Code.Module()  # may be overwritten (not added to) below
+      macIterCode = Code.Module()
+      waitLWCode = Code.Module()
+      syncCode = Code.Module()
+
+      for iui in range(0,kernel["InnerUnroll"]):
+        if self.enable["LocalRead"]:
```

Comment:
Looks like whole for loop is skipped if LocalRead is disabled, maybe swap these lines for clarity?

Reply:
noGlobalLoadLoop is a copy of main loop without globalReads.
noLoadLoop is loop without globalReads, localWrites, pointerSwap, offsetReset and localReads for next loop.

reusing noLoadLoop for noGlobalLoadLoop done.
swap lines for clarity done.